File: src/oop.py
class Rectangle:
    def __init__(self, width, height, color):
        self.__width = width
        self.__height = height
        self.__color = color
    
    def __del__(self):
        print("Destructor")
    def print(self):
        print('Width:', self.__width)
        print('Height:', self.__height)
        print('Color:', self.__color)


    def getWidth(self):
        return self.__width
    def setWidth(self, width):
        self.__width = width
    def getHeight(self):
        return self.__height
    def setHeight(self, height):
        self.__height = height
    def getColor(self):
        return self.__color
    def setColor(self, color):
        self.__color = color
    def getPerimeter(self):
        return (self.__width + self.__height) * 2
    def getArea(self):
        return self.__width * self.__height

# figure1 =Rectangle() #Lỗi
figure2 = Rectangle(5, 10, 'red') #Đúng
# figure1.print()
# figure1._height = 5 #Đúng

figure2.sta = 2
print(Rectangle.sta)
print(figure2.sta)
Rectangle.sta = 5
print(Rectangle.sta)
print(figure2.sta)

File: src/oop.cpp
#include <iostream>
#include <string>
using namespace std;

class Rectangle {
private:
    int width;
    int height;
    string color;
public: 
    int getWidth();
    void setWidth(int width);
    int getHeight();
    void setHeight(int height);
    string getColor();
    void setColor(string color);
    int getPerimeter();
    int getArea();
};

int Rectangle::getWidth() {
    return width;
}
void Rectangle::setWidth(int width) {
    this->width = width;
}
int Rectangle::getHeight() {
    return height;
}
void Rectangle::setWidth(int width) {
    this->height = height;
}
string Rectangle::getColor() {
    return color;
}
void Rectangle::setColor(string color) {
    this->color = color;
}
int Rectangle::getPerimeter() {
    return (width + height) * 2;
}
int Rectangle::getArea() {
    return width * height;
} 

int main() {
    Rectangle figure1;
    // figure1.width = 1;    Lỗi không thể truy xuất thành phần private
    // figure1.color = "red";
    // cout << "Figure1 color: " << figure1.color << endl;
    figure1.setWidth(10);
    figure1.setHeight(5);
    cout << "Figure1 area: " << figure1.getArea() << endl;
    cout << "Figure1 perimeter: " << figure1.getPerimeter() << endl;

    return 0;
}
